src/index.js

```diff
@@ -1,49 +1,82 @@
-    // cheerio (https://cheerio.js.org/) uses the same api as jQuery (http://jquery.com/)
-    // here I do an Array.from to convert the cheerio fake array to a real js array.
-    const entries = Array.from($('article')).map(article =>
-      parseArticle($, article)
-    )
-    return addData(entries, 'io.cozy.books').then(() =>
-      saveFiles(entries, fields)
-    )
+async function start(fields) {
+  log('info', 'Authenticating ...')
+  await authenticate(fields.login, fields.password)
+  log('info', 'Successfully logged in')
+  log('info', 'Fetching the list of bills')
+  const $ = await request(`${baseUrl}/FacturesConso/Factures`)
+  log('info', 'Parsing bills')
+  const bills = parseBills($)
+  log('info', 'Saving data to Cozy')
+  await saveBills(bills, fields.folderPath, {
+    identifier: 'vinci'
   })
 }
 
-// The goal of this function is to parse a html page wrapped by a cheerio instance // and return an array of js objects which will be saved to the cozy by addData (https://github.com/cozy/cozy-konnector-libs/blob/master/docs/api.md#module_addData)
-// and saveFiles (https://github.com/cozy/cozy-konnector-libs/blob/master/docs/api.md#savefiles)
-function parseArticle($, article) {
-  const $article = $(article)
-  const title = $article.find('h3 a').attr('title')
-  return {
-    title,
-    price: normalizePrice($article.find('.price_color').text()),
-    url: `${baseUrl}/${$article.find('h3 a').attr('href')}`,
-    // when it finds a fileurl attribute, saveFiles will save this file to the cozy with a filename
-    // name
-    fileurl: `${baseUrl}/${$article.find('img').attr('src')}`,
-    filename: `${title}.jpg`
-  }
+async function authenticate(login, password) {
+  await signin({
+    url: `${baseUrl}/Authentification`,
+    formSelector: '#formAuthentication',
+    formData: {
+      'OpusInternetConnexionModel.NumeroClient': login,
+      'OpusInternetConnexionModel.Password': password
+    }
+  })
 }
 
-// convert a price string to a float
-function normalizePrice(price) {
-  return parseFloat(price.trim().replace('£', ''))
+function parseBills($) {
+  const bills = scrape(
+    $,
+    {
+      fileurl: {
+        sel: 'a',
+        attr: 'href',
+        parse: href => `${baseUrl}${href}`
+      },
+      idFacture: {
+        sel: 'td:nth-child(1)'
+      },
+      amount: {
+        sel: 'td:nth-child(5)',
+        parse: amount => parseFloat(amount.replace(' €', '').replace(',', '.'))
+      },
+      originalDate: {
+        sel: 'td:nth-child(3)',
+        parse: date => moment(date, 'MMMM YYYY')
+      },
+      date: {
+        sel: 'td:nth-child(7)',
+        parse: date => moment(date, 'DD/MM/YYYY').toDate()
+      }
+    },
+    '.table tbody tr'
+  )
+
+  return bills.map(bill => {
+    const { originalDate } = bill
+    delete bill.originalDate
+    return {
+      ...bill,
+      vendor: 'Vinci autoroute',
+      currency: '€',
+      filename: `${originalDate.format('YYYY-MM')}-${String(
```

Comment:
[optim] If we have the name of the road, it would be nice to identify a trip.

Reply:
These are monthly bills. It can be related to many roads. And vinci keeps the details of the road only for the current month.